### ISSUE TITLE
How should the training data be served?

### ISSUE THREAD

**Issue by thompsonmj** [Imageomics/bioclip-vector-db#3]
Full-sized images used for training are split between the EOL images on Hugging Face and the iNat21 and BIOSCAN image sets through their own distribution sources.

However, the images were resized to 224x224 prior to training. For this project, it is a requirement that the nearest-neighbor images be presented in the format they were in for training to provide users an accurate representation of what the model 'knows'. This would probably best be done by putting the full `webdataset` formatted set of TAR files into a private Hugging Face space strictly for serving a handful of images at a time for this project rather than as a redistribution method. This should provide random access to individual images by filename. 

Would it also be useful to see the nearest neighbor images in full original resolution? It would be possible using `ratarmount`, which could FUSE mount the contents of each dataset (`.tar.gz` for EOL and iNat and `.zip` for BIOSCAN) to the API server filesystem to enable random access to individual full-sized images as well.

**Comment by egrace479** [comment]
The Hugging Face solution seems reasonable since our `webdataset` has about 200 TAR files and Hugging Face supports the `webdataset` format.

We'll likely want to have the TAR file/<filename> for the keys in the vector database so it knows how to access the images.

**Comment by NetZissou** [comment]
@thompsonmj this issue might be out-of-date. Should we consider closing this issue?

**Comment by thompsonmj** [comment]
Yes, I think we have settled pretty comfortably on the HDF5/Parquet hybrid format.
